### PR TITLE
EP-480: Checkout Decimal Properties

### PR DESCRIPTION
**Pull request by sunday-okpoluaefe**
# 📲 What

Android QA: Checkout Decimal Properties

# 🤔 Why

Convert checkout properties  ``` checkout_add_ons_minimum_usd ```  ``` checkout_bonus_amount_usd ``` ``` checkout_shipping_amount_usd ```  ``` checkout_reward_minimum_usd ```
from float to decimal

# 🛠 How

- Created an extension ``` Double.multiplyRound2Decimal ``` to Double. 
```
fun Double.multiplyRound2Decimal(other: Double, scale: Int = 2) : Double {
    return BigDecimal(this).multiply(BigDecimal(other)).setScale(scale, RoundingMode.HALF_UP).toDouble()
}

```

- Added unit test 

```
fun testMultiply2RoundDecimal() {

        val val1 = 2.0212
        val val2 = 6.0443

        val expected = 12.22

        assertEquals(val1.multiplyRound2Decimal(val2), expected)
    }

```

# 👀 See

Trello, screenshots, external resources?

| Before 🐛 | After 🦋 |
| --- | --- |
|  |  |

# 📋 QA

Instructions for anyone to be able to QA this work.

# Story 📖

https://kickstarter.atlassian.net/browse/EP-480

### CODE REVIEW

File: app/src/main/java/com/kickstarter/viewmodels/BackingFragmentViewModel.kt [hunk 1]
```diff
@@ -388,6 +389,7 @@ interface BackingFragmentViewModel {
                 .compose(bindToLifecycle())
                 .subscribe {
                     this.lake.trackManagePledgePageViewed(it.first, it.second)
+                    Log.e("HERE", "GOTCHA")
```
Comment (leighdouglas) [inline app/src/main/java/com/kickstarter/viewmodels/BackingFragmentViewModel.kt:392]: nit: remove log statement

**Review by leighdouglas — APPROVED** [review]
Looks great! 🎉  Just update the summary of this PR when you have a chance. And three small things:
1. Please remove the log statement before merging
2. Up to you if you want to remove the `multiplyRound2Decimal` extension method, since we aren't using it anymore
3. Should we update the effected properties in `SegmentTest` to make sure we are sending a doubles now? 

Great work on this! 🚀

File: app/src/main/java/com/kickstarter/viewmodels/BackingFragmentViewModel.kt (hunk 1 above)
Comment (sunday-okpoluaefe) [reply app/src/main/java/com/kickstarter/viewmodels/BackingFragmentViewModel.kt]: Done

**Comment by sunday-okpoluaefe** [comment]
> Looks great! 🎉 Just update the summary of this PR when you have a chance. And three small things:
> 
> 1. Please remove the log statement before merging
> 2. Up to you if you want to remove the `multiplyRound2Decimal` extension method, since we aren't using it anymore
> 3. Should we update the effected properties in `SegmentTest` to make sure we are sending a doubles now?
> 
> Great work on this! 🚀

*3 - All tests that send checkout properties automatically use the ``` Double.round() ``` method. All test passed

**Pull request merged**